File: 1040.rs
use std::io;

fn main() {
  let mut input = String::new();

  io::stdin().read_line(&mut input).unwrap();

  let notes: Vec<&str> = input.split_whitespace().collect();

  let n1 = notes[0].parse::<f64>().unwrap();
  let n2 = notes[1].parse::<f64>().unwrap();
  let n3 = notes[2].parse::<f64>().unwrap();
  let n4 = notes[3].parse::<f64>().unwrap();

  let median = ((n1 * 2.0) + (n2 * 3.0) + (n3 * 4.0) + (n4 * 1.0)) / 10.0;

  if median > 6.9 {
    println!("Media: {:.1}
Aluno aprovado.", median);
    return;
  } else if median >= 5.0 && median <= 6.9 {
    let mut input_exam = String::new();
    io::stdin().read_line(&mut input_exam).unwrap();

    let exam = input_exam.trim().parse::<f64>().unwrap();

    let exam_median = (median + exam) / 2.0;
    if exam_median >= 5.0 {
      println!("Media: {:.1}
Aluno em exame.
Nota do exame: {:.1}
Aluno aprovado.
Media final: {:.1}", median, exam, exam_median);
      return ;
    }
  }

    println!("Media: {:.1}
Aluno reprovado.", median);
}

File: Iniciante/1040/1040.java
import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    Scanner scan = new Scanner(System.in);

    String[] values = scan.nextLine().split(" ");

    float n1 = Float.parseFloat(values[0]);
    float n2 = Float.parseFloat(values[1]);
    float n3 = Float.parseFloat(values[2]);
    float n4 = Float.parseFloat(values[3]);

    float median = ((n1 * 2) + (n2 * 3) + (n3 * 4) + (n4 * 1)) / 10.0f;

    System.out.printf("Media: %.1f%n", median);
    if (median >= 7.0) {
      System.out.println("Aluno aprovado.");
    } else if (median >= 5.0 && median < 7.0) {
      System.out.println("Aluno em exame.");
      Float n5 = Float.parseFloat(scan.nextLine());

      System.out.printf("Nota do exame: %.1f\n", n5);
      float final_median = (median + n5) / 2.0f;

      if (final_median >= 5.0) {
        System.out.println("Aluno aprovado.");
      } else {
        System.out.println("Aluno reprovado.");
      }

      System.out.printf("Media final: %.1f\n", final_median);
    } else {
      System.out.println("Aluno reprovado.");
    }
  }
}
